File: redome/src/main/java/br/org/cancer/modred/controller/dto/AnaliseMatchPreliminarDTO.java
package br.org.cancer.modred.controller.dto;

import java.util.List;

import br.org.cancer.modred.model.BuscaPreliminar;

/**
 * DTO utilizado para popular o carrousel de matchs preliminares.
 * 
 * @author Pizão
 */
public class AnaliseMatchPreliminarDTO {
	
	private BuscaPreliminar buscaPreliminar;
	private List<MatchDTO> listaFase1;
	private List<MatchDTO> listaFase2;
	private List<MatchDTO> listaFase3;
	private Long totalMedula;
	private Long totalCordao;
	
	
	public BuscaPreliminar getBuscaPreliminar() {
		return buscaPreliminar;
	}
	public void setBuscaPreliminar(BuscaPreliminar buscaPreliminar) {
		this.buscaPreliminar = buscaPreliminar;
	}
	public List<MatchDTO> getListaFase1() {
		return listaFase1;
	}
	public void setListaFase1(List<MatchDTO> listaFase1) {
		this.listaFase1 = listaFase1;
	}
	public List<MatchDTO> getListaFase2() {
		return listaFase2;
	}
	public void setListaFase2(List<MatchDTO> listaFase2) {
		this.listaFase2 = listaFase2;
	}
	public List<MatchDTO> getListaFase3() {
		return listaFase3;
	}
	public void setListaFase3(List<MatchDTO> listaFase3) {
		this.listaFase3 = listaFase3;
	}
	public Long getTotalMedula() {
		return totalMedula;
	}
	public void setTotalMedula(Long totalMedula) {
		this.totalMedula = totalMedula;
	}
	public Long getTotalCordao() {
		return totalCordao;
	}
	public void setTotalCordao(Long totalCordao) {
		this.totalCordao = totalCordao;
	}

}

File: redome-front/src/app/shared/dto/analise.match.preliminar.dto.ts
import { BuscaPreliminar } from 'app/shared/model/busca.preliminar';
import { BaseEntidade } from "app/shared/base.entidade";
import { MatchDTO } from "app/shared/component/listamatch/match.dto";
import { ConvertUtil } from "app/shared/util/convert.util";

/**
 * Classe que representa o dto com a listade matchs preliminares para uma busca preliniar de um paciente
 * 
 * @author Bruno Sousa
 * @export
 * @class AnaliseMatchPreliminarDTO
 */
export class AnaliseMatchPreliminarDTO extends BaseEntidade {

    private _buscaPreliminar: BuscaPreliminar;
    private _listaFase1: MatchDTO[];
    private _listaFase2: MatchDTO[];
    private _listaFase3: MatchDTO[];

    private _totalMedula: number;
    private _totalCordao: number;
    
    /**
     * Getter buscaPreliniar
     * @return {number}
     */
	public get buscaPreliminar(): BuscaPreliminar {
		return this._buscaPreliminar;
	}

    /**
     * Setter buscaPreliniar
     * @param {number} value
     */
	public set buscaPreliminar(value: BuscaPreliminar) {
		this._buscaPreliminar = value;
	}

	public get listaFase1(): MatchDTO[] {
		return this._listaFase1;
	}

	public set listaFase1(value: MatchDTO[]) {
		this._listaFase1 = value;
	}

	public get listaFase2(): MatchDTO[] {
		return this._listaFase2;
	}

	public set listaFase2(value: MatchDTO[]) {
		this._listaFase2 = value;
	}

	public get listaFase3(): MatchDTO[] {
		return this._listaFase3;
	}

	public set listaFase3(value: MatchDTO[]) {
		this._listaFase3 = value;
	}

    /**
     * Getter totalMedula
     * @return {number}
     */
	public get totalMedula(): number {
		return this._totalMedula;
	}

    /**
     * Setter totalMedula
     * @param {number} value
     */
	public set totalMedula(value: number) {
		this._totalMedula = value;
	}

    /**
     * Getter totalCordao
     * @return {number}
     */
	public get totalCordao(): number {
		return this._totalCordao;
	}

    /**
     * Setter totalCordao
     * @param {number} value
     */
	public set totalCordao(value: number) {
		this._totalCordao = value;
	}

	public jsonToEntity(res: any): AnaliseMatchPreliminarDTO {
        
        if (res.buscaPreliminar) {
            this.buscaPreliminar = new BuscaPreliminar().jsonToEntity(res.buscaPreliminar);
        }
        
        if (res.listaFase1) {
            this.listaFase1 = [];
            res.listaFase1.forEach(matchDTO => {
                this.listaFase1.push(new MatchDTO().jsonToEntity(matchDTO));
            });
        }

        if (res.listaFase2) {
            this.listaFase2 = [];
            res.listaFase2.forEach(matchDTO => {
                this.listaFase2.push(new MatchDTO().jsonToEntity(matchDTO));
            });
        }
        if (res.listaFase3) {
            this.listaFase3 = [];
            res.listaFase3.forEach(matchDTO => {
                this.listaFase3.push(new MatchDTO().jsonToEntity(matchDTO));
            });
        }
    
        this.totalMedula = ConvertUtil.parseJsonParaAtributos(res.totalMedula, new Number());
        this.totalCordao = ConvertUtil.parseJsonParaAtributos(res.totalCordao, new Number());

        return this;
    }
}